src/coreclr/jit/emitarm64.cpp

```diff
@@ -16069,11 +16070,12 @@ bool emitter::IsRedundantLdStr(
         return false;
     }
 
-    regNumber prevReg1   = emitLastIns->idReg1();
-    regNumber prevReg2   = emitLastIns->idReg2();
-    insFormat lastInsfmt = emitLastIns->idInsFmt();
-    emitAttr  prevSize   = emitLastIns->idOpSize();
-    ssize_t prevImm = emitLastIns->idIsLargeCns() ? ((instrDescCns*)emitLastIns)->idcCnsVal : emitLastIns->idSmallCns();
+    regNumber prevReg1   = emitGetLastIns()->idReg1();
```

Comment:
Maybe it would be better to add:
```
    const emitLastIns* = emitGetLastIns();
```
instead of calling the function many times?

Reply:
A general pattern might be:
```
if (!emitHasLastIns())
{
    return;
}
const emitLastIns* = emitGetLastIns();
...
```
or, similarly,
```
if (emitHasLastIns())
{
    const emitLastIns* = emitGetLastIns();
    ...
}
```

Reply:
I think the CPP compilers will do this already right? `emitGetLastIns` is small so it will be inlined and a CSE could be picked up.

Reply:
True, I can fix that.